cpp/src/table/legacy/table.cpp

```diff
@@ -58,8 +58,15 @@ table::table(gdf_size_type num_rows,
                       gdf_column_view_augmented(col, nullptr, nullptr, num_rows,
                                                 dtype, 0, extra_info),
                      "Invalid column parameters");
-
-        RMM_ALLOC(&col->data, cudf::size_of(dtype) * num_rows, stream);
+        
+        // Allocation size should be  aligned to 4 bytes. The use of 
+        // `round_up_safe`guarantee correct execution with cuda-memcheck  
+        // for cases when the sizeof(type) is 1 or 2 bytes. 
+        size_t buffer_size = cudf::size_of(dtype) * num_rows;
+        size_t buffer_size_aligned = 
+            cudf::util::round_up_safe((size_t)buffer_size, (size_t)sizeof(int32_t));
```

Comment:
You could use `round_up_unsafe` here safely since num_rows is int32 and buffer_size is int64 so it can't possibly overflow when rounding up to a multiple of 4...

Reply:
About the last observation. I looked for round_up_unsafe function and I didn't find it in branch-0.10. Which one are you referring to? 

About the `update_nvcategories`. This function is used by sort and hash based groupby.
```
static void update_nvcategories(table const& input_keys, table& output_keys,
                         table const& input_values, table& output_values) {
  nvcategory_gather_table(input_keys, output_keys);
  nvcategory_gather_table(input_values, output_values);
}
```
It uses internally `nvcategory_gather_table` function and it returns an `gdf_error`. So this is what I am thinking.  
```
gdf_error update_err  nvcategory_gather_table(input_keys, output_keys);
CUDF_EXPECTS(update_err == GDF_SUCCESS, "nvcategory_gather_table error for keys");
update_err =  nvcategory_gather_table(input_values, output_values);
CUDF_EXPECTS(update_err == GDF_SUCCESS, "nvcategory_gather_table error for values");
```
Let me known if this is OK?

Reply:
That all looks good, yes.

Reply:
Don't worry about changing the round_up method, I doubt this one is costing any performance.